src/components/date-picker/dropdown.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import {getMonthNames, getMonthFromOffset, getDayNames, getDaysInMonth, getNumOfPreviousDays, getNumOfFollowingDays} from '../../common/date-helpers';
+import {observable, action, computed} from 'mobx';
+import {observer} from 'mobx-react';
+import {Day} from './day';
+const styles = require('./date-picker.st.css').default;
+
+export interface DropdownProps {
+    value: Date;
+    onChange(date: Date): void;
+}
+
+const monthNames = getMonthNames();
+
+@observer
```

Comment:
I still don't see problem with rendering calendar grid and if it has any performance issues. 
no any issues since jquery.